barretenberg/cpp/src/barretenberg/polynomials/eq_polynomial.hpp

```diff
@@ -0,0 +1,249 @@
+// === AUDIT STATUS ===
+// internal:    { status: not started, auditors: [], date: YYYY-MM-DD }
+// external_1:  { status: not started, auditors: [], date: YYYY-MM-DD }
+// external_2:  { status: not started, auditors: [], date: YYYY-MM-DD }
+// =====================
+
+#pragma once
+#include "barretenberg/common/bb_bench.hpp"
+#include "barretenberg/common/compiler_hints.hpp"
+#include "barretenberg/common/thread.hpp"
+#include "barretenberg/numeric/bitop/get_msb.hpp"
+#include "barretenberg/stdlib/primitives/bool/bool.hpp"
+#include "gate_separator.hpp"
+
+#include <cstddef>
+#include <vector>
+namespace bb {
+
+/**
+ * @brief Prover-side eq(X, r) polynomial over Boolean hypercube.
+ *
+ * eq(X, r) = ∏_{i=0}^{d-1} ((1 - r_i)(1 - X_i) + r_i X_i)
+ *          = ∏_{i=0}^{d-1} (b_i + a_i X_i),  where a_i = 2r_i - 1, b_i = 1 - r_i
+ *
+ * This class provides two construction strategies:
+ *
+ * 1. **Optimal method (default)**: Uses coordinate transformation + GateSeparatorPolynomial
+ *    - Transforms eq(X, r) = C · ∏_i (1 + γ_i X_i), where γ_i = r_i / (1 - r_i), C = ∏_i (1 - r_i)
```

Comment:
I think this should be
```suggestion
 *    - Transforms eq(X, r) = C · ∏_i (1 + (γ_i - 1) X_i), where γ_i = r_i / (1 - r_i), C = ∏_i (1 - r_i)
```